### ISSUE TITLE
LightGBM on Apache Spark - support for scala 2.12

### ISSUE THREAD

**Issue by informatica92** [microsoft/SynapseML#907]
Is there a reason why LightGBM on Apache Spark doesn't have a jar for scala 2.12? It would be nice to have support for both versions (2.11 and 2.12)

**Comment by JWenBin** [comment]
can i see your example？plz

**Comment by imatiach-msft** [comment]
Agreed, I've seen some conversations around this and I believe the latest was that there are some dependencies of mmlspark which don't yet support 2.11.  Otherwise, if those are excluded somehow or resolved, I believe it is straightforward.

**Comment by brunocous** [comment]
Maybe an pre-lease containing 2.12 (excluding incompatible dependencies) could be released? Maybe spark 3.0.0 support could also be added to that one?

**Comment by informatica92** [comment]
It would be great in my opinion... I had to downgrade Scala in order to make it work

**Comment by anupam-burra-zocdoc** [comment]
We're holding up upgrading our Runtime framework, blocked on this upgrade, so this would be very helpful. Is anyone able to provide an ETA for when this will happen so we can plan around this better?

**Comment by juanpaulo** [comment]
related to #445 and #912 ?

**Comment by juanpaulo** [comment]
Looks like this is resolved by the recently merged PR: #970

**Comment by dsv-dungnt** [comment]
Does anyone know how to try with the nightly build?

**Comment by nhymxu** [comment]
Does anyone know how to try with the nightly build?

**Comment by imatiach-msft** [comment]
@nhymxu the nightly build is on github with a badge:
![image](https://user-images.githubusercontent.com/24683184/115504237-9fa89200-a245-11eb-9cac-defe35b1852f.png)

**Comment by imatiach-msft** [comment]
latest mmlspark support scala 2.12

**Comment by vinay-singh27** [comment]
I have installed the latest MML Spark version(com.microsoft.ml.spark:mmlspark_2.11:1.0.0-rc3) on databricks with scala 2.12 & spark 3.0.1 but I am unable to run MMLSpark on it. While running, it gives error "java.lang.NoClassDefFoundError: org/apache/spark/ml/util/MLWritable$class". I believe this issue is because of the non-compatibility with Scala 2.12 or is there anything additional required to run it on Scala 2.12?

**Comment by imatiach-msft** [comment]
@vinay-singh27 just from this version: 

com.microsoft.ml.spark:mmlspark_2.11:1.0.0-rc3

you must have installed the 2.11 scala version, not 2.12.  RC3 release does not have 2.12 version, it's only available on master.  However you can install any master build, for example:
coordinate:
com.microsoft.ml.spark:mmlspark_2.12:1.0.0-rc3-62-25d40cff-SNAPSHOT
repository:
https://mmlspark.azureedge.net/maven

you can see latest master in github main page

**Comment by cosmincatalin** [comment]
Is there any chance we'll get the 2.12 version in Maven Central soon?

**Comment by imatiach-msft** [comment]
@cosmincatalin we haven't been releasing to maven central for a while, we have always been releasing to our repository:
https://mmlspark.azureedge.net/maven
all of the rc1-rc3 releases have been there.  Do you have any issues with using the repository instead of maven central?  You should be able to specify the repository when getting spark packages on all clusters/platforms.

**Comment by cosmincatalin** [comment]
That's exactly the problem I have. Since your repository is not standard, it's not mirrored behind my corporate proxy.

**Comment by mhamilton723** [comment]
Just to re-provide an answer for those looking for help

Our latest maven artifact is for 2.12
https://central.sonatype.com/artifact/com.microsoft.azure/synapseml_2.12/0.11.2

**Comment by ranadeepsingh** [comment]
Thanks everyone for raising this and for the follow-up discussion. Scala 2.12 support shipped and is the current supported artifact line: SynapseML v1.1.3 is published as `com.microsoft.azure:synapseml_2.12`.

References:
- Current release: https://github.com/microsoft/SynapseML/releases/tag/v1.1.3
- Maven artifact: https://central.sonatype.com/artifact/com.microsoft.azure/synapseml_2.12

I am closing this as completed. If you encounter a Scala 2.12 compatibility regression with a current SynapseML release, please open a new issue with the SynapseML, Spark, and Scala versions so we can investigate it directly.
